src/main/java/org/fungover/Topic.java

```diff
@@ -0,0 +1,68 @@
+package org.fungover;
+
+import java.util.Objects;
+
+public class Topic {
+    private String name;
+
+    private Topic (String name) {
+        this.name = name;
+   }
+
+   public String getName() {
+        return name;
+   }
+
+    public static Topic create(String topicName) {
+        if (topicName == null || topicName.isEmpty())
+            throw new IllegalArgumentException("Invalid topic: Should contain at least one character");
+        if (topicName.startsWith("/"))
+            throw new IllegalArgumentException("Invalid topic: Can not begin with '/'");
+        if (topicName.contains("#")) {
+            int count = countAmountOfMultiLevelWildcard(topicName);
+            if (count == 1 && !topicName.endsWith("/#") || count > 1)
+                throw new IllegalArgumentException("Invalid topic: Multi-level wildcard (#) must be placed as the last character in the topic, preceded by a forward slash");
+        }
```

Comment:
```suggestion
          public static Topic create(String topicName) {
        errorHandling(topicName);
        return new Topic(topicName);
    }

    private static void errorHandling(String topicName) {
        if (topicName == null || topicName.isEmpty())
            throw new IllegalArgumentException("Invalid topic: Should contain at least one character");
        if (topicName.startsWith("/"))
            throw new IllegalArgumentException("Invalid topic: Can not begin with '/'");
        if (topicName.contains("#")) {
            int count = countAmountOfMultiLevelWildcard(topicName);
            if (count == 1 && !topicName.endsWith("/#") || count > 1)
                throw new IllegalArgumentException("Invalid topic: Multi-level wildcard (#) must be placed as the last character in the topic, preceded by a forward slash");
        }
    }
 
```
In this context, we're delving into personal fields, but the current method appears a bit overloaded to me. It might be more efficient to separate the error handling into its own method.

Reply:
Awesome! Thanks!